### PR TITLE
chore(nix): Update refs to submodules & explain why this is needed

### PR DESCRIPTION
**Pull request by bew**
I saw that you updated harfbuzz' version in ec438e714c25f24783e2a23df5d3fb8eef192b55, and noticed last time that the Nix code currently needs to get the submodules manually, so here you go.
I added some explanation why this whole submodule duplication thing is needed at the moment.

I also preferred to use tags instead of commit hash to be much more readable and maintainable.

cc: @happenslol @gabyx is that okay for you?

### CODE REVIEW

File: nix/flake.nix [hunk 1]
```diff
@@ -12,23 +12,27 @@
...
     };
-
     harfbuzz = {
-      url = "github:harfbuzz/harfbuzz/60841e26187576bff477c1a09ee2ffe544844abc";
+      url = "github:harfbuzz/harfbuzz/8.4.0";
       flake = false;
     };
-
     libpng = {
       url = "github:glennrp/libpng/8439534daa1d3a5705ba92e653eda9251246dd61";
```
Comment (bew) [inline nix/flake.nix:31]: I didn't find a relevant tag for this commit, is that correct @wez 🤔

(I'm curious)
Is there something specific about that commit? or it was the last when you included that submodule and never needed to update?

**Comment by happenslol** [comment]
Looks good to me, I went through the threads you linked when I created the flake 😅

Linking the tags instead of commit hashes makes sense for readability. I'm worried that the manual upgrade process becomes harder/more tedious though, since now we need to look through the repos and find the tag instead of just using the ref the submodule points to.

**Comment by bew** [comment]
> I went through the threads you linked when I created the flake 😅


yeah I figured, now it's explicit and we can subscribe to something to remove this 'hack' some time in the future (lazy trees is actively being worked on atm, according to recent nix team meeting summaries 👌)

> I'm worried that the manual upgrade process becomes harder/more tedious though, since now we need to look through the repos and find the tag instead of just using the ref the submodule points to.

Well, running `git submodule status` seems to show the result of `git describe` for each submodules, so it helps a lot:
image

File: nix/flake.nix (hunk 1 above)
Comment (wez) [reply nix/flake.nix:35]: I have no specific recollection here. It's possible that updating to a newer rev would be a good idea.

File: nix/flake.nix [hunk 2]
```diff
@@ -12,23 +12,31 @@
...
+    #
+    # ... In the meantime we kinda duplicate the dependencies here then replace the submodules with
+    # links to each repo in package sources.
+    #
+    # Try to use tags when possible to increase readability
+    # (note: `git submodule status` in wezterm repo will show the `git describe` result for each
+    # submodule, can help finding a tag if any)
     freetype2 = {
-      url = "github:wez/freetype2/de8b92dd7ec634e9e2b25ef534c54a3537555c11";
+      url = "github:wez/freetype2/e4586d960f339cf75e2e0b34aee30a0ed8353c0d";
```
Comment (wez) [inline nix/flake.nix:27]: FWIW, you could use the tag for this too:

```
 e4586d960f339cf75e2e0b34aee30a0ed8353c0d deps/freetype/freetype2 (VER-2-13-1)
```

Comment (bew) [reply nix/flake.nix:27]: hmm I don't see that tag 👀
https://github.com/wez/freetype2/tags

**Comment by wez** [comment]
Thanks!

**Pull request merged**